clipper_admin/clipper_admin/docker/docker_container_manager.py

```diff
@@ -170,16 +212,49 @@ def start_clipper(self,
         run_query_frontend_metric_image(
             query_frontend_metric_name, self.docker_client, query_name,
             self.common_labels, self.extra_container_kwargs)
-        setup_metric_config(query_frontend_metric_name,
+
+        self.prom_config_path = tempfile.NamedTemporaryFile(
+            'w', suffix='.yml', delete=False).name
+        self.prom_config_path = os.path.realpath(
+            self.prom_config_path)  # resolve symlink
+        setup_metric_config(query_frontend_metric_name, self.prom_config_path,
                             CLIPPER_INTERNAL_METRIC_PORT)
-        run_metric_image(self.docker_client, self.common_labels,
-                         self.prometheus_port, self.extra_container_kwargs)
+
+        self.prometheus_port = find_unbound_port(self.prometheus_port)
+        metric_labels = self.common_labels.copy()
+        metric_labels[CLIPPER_DOCKER_PORT_LABELS['metric']] = str(
+            self.prometheus_port)
+        metric_labels[CLIPPER_METRIC_CONFIG_LABEL] = self.prom_config_path
+        run_metric_image(self.docker_client, metric_labels,
+                         self.prometheus_port, self.prom_config_path,
+                         self.extra_container_kwargs)
 
         self.connect()
 
     def connect(self):
-        # No extra connection steps to take on connection
-        return
+        """
+        Use the cluster name to update ports. Because they might not match as in
+        start_clipper the ports might be changed.
+        :return: None
+        """
+        containers = self.docker_client.containers.list(
+            filters={
+                'label':
+                ['ai.clipper.container.label={}'.format(self.cluster_name)]
```

Comment:
Don't use the literal value of the label, use a variable to reference it instead

Reply:
Reminder